x-pack/plugins/security_solution/server/endpoint/routes/actions/audit_log_handler.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TypeOf } from '@kbn/config-schema';
+import { RequestHandler } from 'kibana/server';
+import { AGENT_ACTIONS_INDEX } from '../../../../../fleet/common';
+import { EndpointActionLogRequestSchema } from '../../../../common/endpoint/schema/actions';
+
+import { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { EndpointAppContext } from '../../types';
+
+export const actionsLogRequestHandler = (
+  _endpointContext: EndpointAppContext
+): RequestHandler<
+  TypeOf<typeof EndpointActionLogRequestSchema.params>,
+  unknown,
+  unknown,
+  SecuritySolutionRequestHandlerContext
+> => {
+  return async (context, req, res) => {
+    const esClient = context.core.elasticsearch.client.asCurrentUser;
+    let result;
+    try {
+      result = await esClient.search({
+        index: AGENT_ACTIONS_INDEX,
+        body: {
+          query: {
+            match: {
+              agents: req.params.agent_id,
+            },
+          },
+        },
+      });
+    } catch (error) {
+      return res.customError({
+        statusCode: 500,
+        body: {
+          message: error,
+        },
+      });
+    }
+    if (result?.statusCode !== 200) {
+      return res.customError({
+        statusCode: 500,
+        body: {
+          message: `Error fetching Actions Log for: ${req.params.agent_id}`,
```

Comment:
clarify message here to indicate problem getting actions for agent id (so that consumer of API understands we are using agent id.

Reply:
done 2dd22ed92b3c54da9bdcce467dbe2d7e87097e4d